EduroamConfigure/ProfileXml.cs

```diff
@@ -379,8 +379,8 @@ public static bool SupportsHs2(EapConfig.AuthenticationMethod authMethod)
             // TODO: hotspot2.0 requires Windows 10
             bool hasOID = authMethod.EapConfig.CredentialApplicabilities
                 .Any(cred => cred.ConsortiumOid != null);
-            bool isPEAP = authMethod.EapType == EapType.PEAP;
-            return hasOID && !isPEAP; // Hotstpot2.0 does not support PEAP
+            // TODO: Are there any authentication methods that cannot work on HS20?
+            return hasOID && UserDataXml.IsSupported(authMethod);
```

Comment:
This is a bit misleading, since the ProfileXml.IsSupported() only accounts for WLANProfile, while ProfileXml.SupportsHs2() now checks both for WLANProfile and EapUserData. The split responsibility between the two classes are now broken. 

I'd rather move this check to
https://github.com/geteduroam/eduroam-supplicant-win/blob/main/EduroamConfigure/EapConfig.cs#L108 .
Something like `.Where(UserData.IsSupported)`

Was it intentional to allow Hs2 for EapType.PEAP? The WLANProfile schema allows it, but that is probably an oversight?

Reply:
That change seems sensible.
As for PEAP, I discussed with @pauldekkers and he doesn't see a reason why it shouldn't work. We know PEAP cannot be mentioned in NAI realms, and Apple completely rejects the notion of PEAP in HS20. But nothing in the protocol stops it from being attempted.

However, I did make this change in a separate branch and put it in a PR because I wanted to test this first. We have HS20 at the office and I tested this build today. It works fine. So we're dropping the PEAP check.

Reply:
Addressed in 44ddb57